dali/operators/reader/loader/video/frames_decoder.cc

```diff
@@ -134,6 +135,23 @@ void FramesDecoder::BuildIndex() {
   Reset();
 }
 
+void FramesDecoder::DetectVfr() {
+  if (NumFrames() < 3) {
+    is_vfr_ = false;
+    return;
+  }
+
+  int pts_step = index_[1].pts - index_[0].pts;
+  for (int frame_id = 2; frame_id < NumFrames(); ++frame_id) {
```

Comment:
How fast is that? Can't we do this when the index is created?

Reply:
We could mix it with index creation. For now, I decided to do it like this for readability (look PR teamplate). I added task to optimize it further, when we have proper benchmark. Part of this optimization will be to get this info without any decoding when possible.

Reply:
Understood.